### PR TITLE
patch time.time in historical_connections to avoid windows failures.

### PR DESCRIPTION
**Pull request by jamadden**
Untested locally, copy-n-pasted from DemoStorage.test. Let's see what Travis says.

### CODE REVIEW

**Comment by jamadden** [comment]
Well now they're all failing consistently with the original error (got `[False,False,False,True]` instead of `[False,False,True,True]`).

I think the problem is that the code is assuming that the two actions in line 233 in this block happens simultaneously:

    >>> db2.setHistoricalTimeout(.01)
    >>> time.sleep(.1)
    >>> conns2[2].close(); conns2[3].close() # 233

If the time increments more than .01 between the two `close` calls, then `conns2[2]` gets destroyed when `conns2[3]` gets closed.

Let me try reducing the interval at which `time.time` increments, I guess.

**Comment by jamadden** [comment]
Adjusting the timing exposed a real bug in `availableGC` for historical pools, which only seems to be exposed on Python >= 3.4. Fixed in b68ac8e.

**Comment by tseaver** [comment]
> Adjusting the timing exposed a real bug in availableGC for historical pools, which only seems to be exposed on Python >= 3.4. Fixed in b68ac8e.

W00t!

**Comment by jamadden** [comment]
Should I squash the commits? GitHub can now do that on merge automatically, too.

**Comment by tseaver** [comment]
Squashing should be fine, I think.  The feature is enabled on this repository:  you get a choice between the traditional merge commit and a squash merge on the "confirm merge" button.

**Comment by jamadden** [comment]
Does that count as a "LGTM" (i.e., should I go ahead and merge)?

**Comment by tseaver** [comment]
LGTM :)

**Comment by jimfulton** [comment]
This change broke the test on Mac OS X.  Investigating. :(

File: src/ZODB/historical_connections.txt [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
+    ...    global _now
+    ...    _now += amt
+    >>> if isinstance(time,type):
+    ...    time.time = staticmethod(faux_time_time) # Jython
+    ...    time.sleep = faux_time_sleep
+    ... else:
+    ...     time.time = faux_time_time
+    ...     time.sleep = faux_time_sleep
+    >>> def utcnow():
+    ...     return datetime.datetime.fromtimestamp(_now)
```
Comment (jimfulton) [inline src/ZODB/historical_connections.txt]: datetime.fromtimestamp is defined to return localtime.

**Comment by jimfulton** [comment]
The failure should occur on any machine who's system time isn't UTC.

It looks like a fix is straightforward. I'll make a new pr.

**Pull request merged**